app.rb

```diff
@@ -0,0 +1,160 @@
+  end
+
+  def start
+    puts 'Welcome to the School Library App!'
+    loop do
+      display_menu
+      choice = gets.chomp
+
+      if MENU_OPTIONS.include?(choice)
+        send(MENU_OPTIONS[choice])
+      else
+        puts 'Invalid choice. Please try again.'
+      end
+    end
+  end
+
+  def display_menu
+    puts 'Please choose an option:'
+    MENU_OPTIONS.each do |key, value|
+      puts "#{key}. #{value.to_s.tr('_', ' ')}"
+    end
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'No books available. Please add some books and try again.'
+    else
+      puts 'List of available books:'
+      @books.each_with_index do |book, index|
+        puts "#{index + 1}) Title: #{book.title} | Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'No people added yet. Please add a person and try again.'
+    else
+      puts 'List of people:'
+      @people.each_with_index do |person, index|
+        puts "#{index + 1}) [#{person.class}] Name: #{person.name} | Age: #{person.age} | ID: #{person.id}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Would you like to create a student (1) or a teacher (2)? Select a number: '
+    choice = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    case choice
+    when '1'
+      create_student(name, age)
+    when '2'
+      create_teacher(name, age)
+    else
+      puts 'Invalid choice. Please try again.'
+    end
+  end
+
+  def create_student(name, age)
+    classroom = nil
+    student = Student.new(age, classroom, name)
+    @people << student
+    puts "Student created successfully. ID is #{student.id}"
+  end
+
+  def create_teacher(name, age)
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(age, name, specialization) # Corrected the order of arguments
+    @people << teacher
+    puts "Teacher created successfully. Teacher ID is #{teacher.id}"
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+
+    if title.strip != '' && author.strip != ''
+      book = Book.new(title, author)
+      @books << book
+      puts 'Book created successfully.'
+    else
+      puts 'Please enter the book title and author.'
+    end
+  end
+
+  def create_rental
+    if @books.empty? || @people.empty?
+      puts 'Nothing to see here. Add books and people first.'
+    else
+      puts 'Select a book to rent:'
+      list_books
+      book_number = gets.chomp.to_i - 1
+      puts 'Enter your ID to rent the book:'
+      list_people
+      person_id = gets.chomp.to_i
+      individual = @people.find { |person| person.id == person_id }
+
+      print 'Enter the rental date [yyyy-mm-dd]: '
+      date = gets.chomp.to_s
+      rental = Rental.new(date, @books[book_number], individual)
+      @rentals << rental
+      puts 'Book rented successfully.'
+    end
+  end
```

Comment:
- [ ] Good job creating the UI for the school library 👏 . The UI allows creating book and index and to fetch data however when I tried to create a rental , It is throwing the following exception. Kindly fix that 😄 

![Capture](https://github.com/Eccb7/school-library/assets/80458199/c76b1c6c-042d-4089-814d-51f80d6c2f21)